sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -514,7 +514,8 @@ object ShowTables {
   def getOutputAttrs: Seq[Attribute] = Seq(
     AttributeReference("namespace", StringType, nullable = false)(),
     AttributeReference("tableName", StringType, nullable = false)(),
-    AttributeReference("isTemporary", BooleanType, nullable = false)())
+    AttributeReference("isTemporary", BooleanType, nullable = false)(),
+    AttributeReference("isView", BooleanType, nullable = false)())
```

Comment:
A new column `isView: Boolean` is more efficient, but I'm wondering if a new column `tableType: String` is more user-friendly. The value can be `TABLE` or `VIEW`.

Reply:
Usually, most modern databases store `tableType` as a string column in `INFORMATION_SCHEMA.TABLES`.

On the reading side, they also return a string value for it.

FYI, https://docs.google.com/spreadsheets/d/1LeHYbGCDjgr-rYwQMlHBhJbeDxjOVuUatozgvKdxq6Y/edit#gid=0

Besides, as `SHOW TABLES` is not ANSI standard, so it might be good for us to follow the JDBC standard. Then we can make our command and JDBC meta operation consistent

Reply:
JDBC protocol also use table type, right?

Reply:
Yes, another benefit for using string is if we decide to subdivide tables to something like `SYSTEM TABLE`/`BASE TABLE`, `xxx TABLE` e.t.c, we won't break then